cirq-core/cirq/work/sampler_test.py

```diff
@@ -266,6 +267,30 @@ def test_sampler_run_batch_bad_input_lengths():
         )
 
 
+@duet.sync
+@mock.patch('duet.pstarmap_async')
+@pytest.mark.parametrize('call_count', [1, 2, 3])
+async def test_run_batch_async_sends_circuits_in_chunks(mock, call_count):
```

Comment:
Can you add a test that verifies that the multiple chunks aren't all running at once? e.g., check that  the no calls are made with chunk N before check N-1 completes. That's an important requirement for this feature.

Reply:
Added